modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/RexUtils.java

```diff
@@ -592,6 +595,19 @@ private static Map<Integer, List<RexCall>> mapPredicatesToFields(RexNode conditi
         return res;
     }
 
+    /** */
```

Comment:
```
    private static RexNode expandBooleanFieldComparison(RexNode rexNode, RexBuilder builder) {
        if (rexNode instanceof RexSlot)
            return builder.makeCall(SqlStdOperatorTable.EQUALS, rexNode, builder.makeLiteral(true));
        else if (rexNode instanceof RexCall && rexNode.getKind() == NOT && 
            (rexNode = ((RexCall)rexNode).getOperands().get(0)) instanceof RexSlot)
            return builder.makeCall(SqlStdOperatorTable.EQUALS, rexNode, builder.makeLiteral(false));

        return rexNode;
```
It seems that it is less verbose

Reply:
In this case modified `rexNode` will be returned for `NOT(...)` conditions, where operand of `NOT` is not `RexSlot`